src/topology/mod.rs

```diff
@@ -937,6 +949,69 @@ impl Clone for Topology {
     }
 }
 
+impl Debug for Topology {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut debug = f.debug_struct("Topology");
+
+        // Topology building properties
+        debug
+            .field("is_abi_compatible", &self.is_abi_compatible())
+            .field("build_flags", &self.build_flags())
+            .field("is_this_system", &self.is_this_system())
+            .field("feature_support", self.feature_support());
+        let type_filters = enum_iterator::all::<ObjectType>()
+            .map(|ty| {
+                (
+                    format!("{ty}"),
+                    self.type_filter(ty)
+                        .expect("should always succeed when called with a valid type"),
+                )
+            })
+            .collect::<BTreeMap<_, _>>();
+        debug.field("type_filter", &type_filters);
+
+        // TopologyObject hierarchy
+        let objects_per_depth = NormalDepth::iter_range(NormalDepth::MIN, self.depth())
+            .map(Depth::from)
+            .chain(Depth::VIRTUAL_DEPTHS.iter().copied())
+            .filter_map(|depth| {
+                let objs = self.objects_at_depth(depth).collect::<Vec<_>>();
+                (!objs.is_empty()).then_some((format!("{depth}"), objs))
+            })
+            .collect::<Vec<_>>();
+        debug
+            // Contains the info from most other topology queries
+            .field("objects_per_depth", &objects_per_depth)
+            .field("memory_parents_depth", &self.memory_parents_depth());
+
+        // CPU and node sets of the entire topology
+        debug
+            .field("cpuset", &self.cpuset())
+            .field("complete_cpuset", &self.complete_cpuset())
+            .field("allowed_cpuset", &self.allowed_cpuset())
+            .field("nodeset", &self.nodeset())
+            .field("complete_nodeset", &self.complete_nodeset())
+            .field("allowed_nodeset", &self.allowed_nodeset());
+
+        // Object distances
+        debug.field("distances", &self.distances(None));
+
+        // Memory attributes
+        #[cfg(feature = "hwloc-2_3_0")]
+        {
+            debug.field("builtin_memory_attributes", &self.dump_builtin_attributes());
+        }
+
+        // Kinds of CPU cores
+        #[cfg(feature = "hwloc-2_4_0")]
+        {
+            let cpu_kinds = self.cpu_kinds().into_iter().flatten().collect::<Vec<_>>();
+            debug.field("cpu_kinds", &cpu_kinds);
+        }
+        debug.finish()
+    }
+}
+
```

Comment:
Needed a Topology Debug and PartialEq impl to easily check that nothing has changed, so here they are.